application/F3DStarter.cxx

```diff
@@ -700,6 +708,123 @@ void F3DStarter::Render()
   f3d::log::debug("Render done");
 }
 
+//----------------------------------------------------------------------------
+void F3DStarter::SaveScreenshot(const std::string& filenameTemplate)
+{
+
+  const auto getScreenshotDir = []()
+  {
+    for (const char* const& candidate : { "XDG_PICTURES_DIR", "HOME", "USERPROFILE" })
+    {
+      char* val = std::getenv(candidate);
+      if (val != nullptr)
+      {
+        std::filesystem::path path(val);
+        if (std::filesystem::is_directory(path))
+        {
+          return path;
+        }
+      }
+    }
+
+    return std::filesystem::current_path();
+  };
+
+  const auto applyFilenameTemplate = [&](const std::string& fn_template)
+  {
+    std::string filename = fn_template;
+
+    {
+      std::regex re("\\{app\\}");
+      filename = std::regex_replace(filename, re, F3D::AppName);
+    }
+    {
+      std::regex re("\\{version\\}");
+      filename = std::regex_replace(filename, re, F3D::AppVersion);
+    }
+    {
+      std::regex re("\\{model\\}");
+      if (std::regex_search(filename, re))
+      {
+        const auto file = this->Internals->FilesList[this->Internals->CurrentFileIndex];
+        filename = std::regex_replace(filename, re, file.filename().string());
+      }
+    }
```

Comment:
how about adding support for file numbering only when using a `{n}` semantic ?

Reply:
can be done yes, would probably need to be on by default then?